### PR TITLE
add optional maxAge argument to load method

### LINKED ISSUE #3: Load objects for maxAge
Hello @onmyway133 👋,

would you like to see a possibility to load objects related to a maximum age? I like your project and the simplicity but would like to see a function like this:

```swift
func load<T: Codable>(forKey key: String, as: T.Type, maxAge: TimeInterval? = nil)
```
The idea is to use the `modificationDate` and the current `Date()` as references.

I ask in advanced before I put effort in the implementation and you don't like to see something like it.

### PR DESCRIPTION
**Pull request by mrgrauel**
Hi @onmyway133 👋 

I hope you like to but I wasn't able to understand your testing setup. Couldn't get the tests running that's why I didn't add any. Can you help me out with the setup?

Closes #3

### CODE REVIEW

**Comment by onmyway133** [comment]
@mrgrauel Hi, thanks for the PR, I've made some comments

File: Sources/Storage+Codable.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ public extension Storage {
         })
     }
 
-    func load<T: Codable>(forKey key: String, as: T.Type) throws -> T {
+    func load<T: Codable>(forKey key: String, as: T.Type, maxAge: TimeInterval = .greatestFiniteMagnitude) throws -> T {
```
Comment (onmyway133) [inline Sources/Storage+Codable.swift]: This is great, I also want it to be like a enum Expiry with `never` and `date` option

File: Sources/Storage+Info.swift [hunk 2]
```diff
@@ -59,4 +44,26 @@ public extension Storage {
...
+        let fileUrl = self.fileUrl(forKey: key)
+        let attributes = try fileManager.attributesOfItem(atPath: fileUrl.path)
+        let modificationDate: Date? = attributes[.modificationDate] as? Date
+        let size: UInt64? = attributes[.size] as? UInt64
+
+        return File(
+            name: key,
+            url: fileUrl,
+            modificationDate:
+            modificationDate,
```
Comment (onmyway133) [inline Sources/Storage+Info.swift]: Parameter should be in same line

File: Sources/Storage.swift [hunk 3]
```diff
@@ -91,7 +93,15 @@ extension Storage {
             .trueOrThrow(StorageError.createFile)
     }
 
-    func commonLoad<T>(forKey key: String, fromData: (Data) throws -> T) throws -> T {
+    func commonLoad<T>(forKey key: String, withMaxAge maxAge: TimeInterval, fromData: (Data) throws -> T) throws -> T {
+        func verify(maxAge: TimeInterval, forKey key: String) throws -> Bool {
+            Date().timeIntervalSince(try modificationDate(forKey: key)) <= maxAge
```
Comment (onmyway133) [inline Sources/Storage.swift]: Maybe we can have `Date()` to be a closure on Storage, so we can unit test it

File: Sources/Storage+Info.swift (hunk 2 above)
Comment (mrosberghaus) [reply Sources/Storage+Info.swift]: f6ec076c554da990c348fa126308d99257553def

File: Sources/Storage+Codable.swift (hunk 1 above)
Comment (mrosberghaus) [reply Sources/Storage+Codable.swift]: Very good idea 👍 20102cd86d4df3d9786a00e0982ef0c078b3e575

Comment (mrosberghaus) [reply Sources/Storage+Codable.swift]: just didn't use a date object. I prefer a `TimeInterval` for the following reason:
* request shouldn't be older as 5 minutes (TimeInterval: 300)

File: Sources/Storage.swift (hunk 3 above)
Comment (mrosberghaus) [reply Sources/Storage.swift]: I'm not sure what you meant but I hope that's what is on your mind. 20102cd86d4df3d9786a00e0982ef0c078b3e575

File: Sources/Storage+Codable.swift (hunk 1 above)
Comment (onmyway133) [reply Sources/Storage+Codable.swift]: Oh ja, I meant time interval too

**Comment by onmyway133** [comment]
@mrgrauel Looks good to me, do you want to add unit tests in this PR too?

**Review by onmyway133 — APPROVED** [review]

**Comment by mrgrauel** [comment]
@onmyway133 as I mentioned in the description I cannot get the tests running. Can you help me with the setup?

**Comment by onmyway133** [comment]
@mrgrauel oh ja, I think  the problem is I put tests in EasyStashTests because this was supporting CocoaPods, not in the Tests folder like Swift Package Manager does

**Comment by mrgrauel** [comment]
I cannot get it running. If you can please commit.

**Comment by onmyway133** [comment]
@mrgrauel I will merge this for now, will update Tests for SPM, gonna remove CocoaPods

**Comment by onmyway133** [comment]
@magauran Updated tests and make new releases, thanks for your contribution https://github.com/onmyway133/EasyStash/releases/tag/1.1.4

**Pull request merged**